### PR TITLE
CATD-429 Display banner image in Sample App

### PR DESCRIPTION
**Pull request by You-Hsuan**

### CODE REVIEW

File: SampleApp/SampleApp/Screens/Components/BannerView.swift [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            subtitle: dataModel.subtitle,
+            imageUrl: dataModel.imageUrl,
+            linkHandler: dataModel.generateLinkHandler(openHandler: { url in
+                UIApplication.shared.open(url, options: [:])
+            })
+        )
+    }
+}
+
+// You could create and use your own Fetcher instead of using our default:
```
Comment (rossrossp) [inline SampleApp/SampleApp/Screens/Components/BannerView.swift:83]: Lines 83-93 should stay in the CreatablesView

**Comment by rossrossp** [comment]
image

I am seeing a bunch of layout constraint issues ^

File: SampleApp/SampleApp/Screens/Components/BannerView.swift [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                        .frame(minWidth: 0, maxWidth: .infinity, idealHeight: idealImageHeight, alignment: .center)
+                        .clipped()
+                } placeholder: {
+                    Color.gray
+                        .frame(minWidth: 0, maxWidth: .infinity, idealHeight: idealImageHeight)
+                }
+            } else {
+                if let image = imageLoader.image {
+                    Image(uiImage: image)
+                        .resizable()
```
Comment (rossrossp) [inline SampleApp/SampleApp/Screens/Components/BannerView.swift:50]: should it also be scaledToFill and clipped like the AsyncImage?

**Review by paleksandrs — APPROVED** [review]

File: SampleApp/SampleApp/Screens/Components/BannerView.swift (hunk 2 above)
Comment (You-Hsuan) [reply SampleApp/SampleApp/Screens/Components/BannerView.swift:50]: Yes, fixed it, thank you!

File: SampleApp/SampleApp/Screens/Components/BannerView.swift (hunk 1 above)
Comment (You-Hsuan) [reply SampleApp/SampleApp/Screens/Components/BannerView.swift]: Done!

**Comment by You-Hsuan** [comment]
> I am seeing a bunch of layout constraint issues ^

It looks like it's because the `NavigationView` component in ModulesView, and it's strange that the issue only appears on iOS15+ device. But by adding  `.navigationViewStyle(.stack)` for NavigationView, we can resolve this constraints warnings. Thank you!

**Comment by rossrossp** [comment]
Unless I'm going blind, I can't ind the code I said to move back to CreatablesView! Where is it?

**Comment by You-Hsuan** [comment]
> Unless I'm going blind, I can't ind the code I said to move back to CreatablesView! Where is it? (It was the code showing how to create a new Creatable type and corresponding Fetcher).

@rossrossp It's in the line 68-78 in CreatablesView.swift. You could find them by expanding the whole file of CreatablesView.swift.

screenshots0411

File: SampleApp/SampleApp/Screens/ModulesView.swift [hunk 3]
```diff
@@ -32,19 +33,22 @@ struct ModulesView: View {
...
             .navigationBarTitle("Realife Tech SDK")
         }
+        .navigationViewStyle(.stack)
     }
-}
 
-struct ModulesView_Previews: PreviewProvider {
-    static var previews: some View {
-        ModulesView()
+    private func createRLTViewFetcher() -> RLTViewFetcher {
```
[the remark this answers is not in the document]
Comment (You-Hsuan) [reply SampleApp/SampleApp/Screens/ModulesView.swift]: Fixed!

**Review by rossrossp — APPROVED** [review]

**Pull request merged**